src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+        when(oldCartDiscount.isRequiringDiscountCode()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isRequiresDiscountCode()).thenReturn(true);
+
+        final Optional<UpdateAction<CartDiscount>> changeRequiresDiscountCodeUpdateAction =
+                buildChangeRequiresDiscountCodeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeRequiresDiscountCodeUpdateAction).contains(ChangeRequiresDiscountCode.of(true));
+    }
+
+    @Test
+    void buildChangeRequiresDiscountCodeUpdateAction_WithBothReqDisCodeHaveNullValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.isRequiringDiscountCode()).thenReturn(true);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.isRequiresDiscountCode()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeRequiresDiscountCodeUpdateAction =
+                buildChangeRequiresDiscountCodeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeRequiresDiscountCodeUpdateAction).contains(ChangeRequiresDiscountCode.of(false));
+    }
+
+    @Test
+    void buildChangeStackingModeUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getStackingMode()).thenReturn(StackingMode.STOP_AFTER_THIS_DISCOUNT);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getStackingMode()).thenReturn(StackingMode.STACKING);
+
+        final Optional<UpdateAction<CartDiscount>> changeStackingModeUpdateAction =
+                buildChangeStackingModeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeStackingModeUpdateAction)
+                .contains(ChangeStackingMode.of(StackingMode.STACKING));
+    }
+
+    @Test
+    void buildChangeStackingModeUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getStackingMode()).thenReturn(StackingMode.STOP_AFTER_THIS_DISCOUNT);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getStackingMode()).thenReturn(StackingMode.STOP_AFTER_THIS_DISCOUNT);
+
+        final Optional<UpdateAction<CartDiscount>> changeStackingModeUpdateAction =
+                buildChangeStackingModeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeStackingModeUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildChangeStackingModeUpdateAction_WithJustNewStackingModeHasNullValue_ShouldBuildUpdateAction() {
```

Comment:
`WithOnlyNullNewStackingMode`

Reply:
what if old value had `StackingMode.STACKING`?

Reply:
c03a57e

Reply:
but why did you remove also the previous case when the old value has `StackingMode.STOP_AFTER_THIS_DISCOUNT`

Reply:
🤔 It would be a different test, because if the new value in the draft is not set (null) means it's `Stacking`, I will add these test too, it could be helpful for the active, require discount code case.

Reply:
5bd6ba4